packages/algob/src/lib/tx.ts

```diff
@@ -141,6 +143,42 @@ async function sendAndWait (
   return await deployer.waitForConfirmation(txInfo.txId);
 }
 
+/**
+ * Make transaction parameters and update ASA and SSC params
+ * @param deployer Deployer object
+ * @param t Execution parameters
+ * @param index index of current execParam
+ * @param txIdxMap Map for index to name
+ */
+async function mkTx (
+  deployer: Deployer,
+  txn: rtypes.ExecParams,
+  index: number,
+  txIdxMap: Map<number, string>
+): Promise<Transaction> {
+  switch (txn.type) {
+    case TransactionType.DeployASA: {
+      deployer.assertNoAsset(txn.asaName);
+      txn.asaDef = deployer.getASADef(txn.asaName);
```

Comment:
Please update this to handle if user itself passes values in `execParams.asaDef` (i think we'll need to use `overrideASADef()` here). 
Something like:
```javascript
txn.asaDef  = overrideASADef(deployer.getaccountsByName(), deployer.getASADef(txn.asaName), txn.asaDef);
```

Reply:
this depends if overrideASADef PR is merged